matpower/matpower-converter/src/main/java/com/powsybl/matpower/converter/MatpowerExporter.java

```diff
@@ -0,0 +1,419 @@
+                        targetV *= vl.getNominalV() / regulatedBus.getVoltageLevel().getNominalV();
+                        LOGGER.warn("Generator remote voltage control not supported in Matpower model, rescale targetV of '{}' from {} to {}",
+                                g.getId(), oldTargetV, targetV);
+                    }
+                    mGen.setVoltageMagnitudeSetpoint(targetV);
+                } else {
+                    mGen.setVoltageMagnitudeSetpoint(0);
+                }
+                mGen.setMinimumRealPowerOutput(g.getMinP());
+                mGen.setMaximumRealPowerOutput(g.getMaxP());
+                mGen.setMinimumReactivePowerOutput(g.getReactiveLimits().getMinQ(g.getTargetP()));
+                mGen.setMaximumReactivePowerOutput(g.getReactiveLimits().getMaxQ(g.getTargetP()));
+                model.addGenerator(mGen);
+            }
+        }
+
+        createDanglingLineGenerators(network, model, context);
+    }
+
+    private static int getBranchCount(Bus bus) {
+        int[] branchCount = new int[1];
+        bus.visitConnectedEquipments(new DefaultTopologyVisitor() {
+            @Override
+            public void visitLine(Line line, Branch.Side side) {
+                branchCount[0]++;
+            }
+
+            @Override
+            public void visitTwoWindingsTransformer(TwoWindingsTransformer transformer, Branch.Side side) {
+                branchCount[0]++;
+            }
+
+            @Override
+            public void visitThreeWindingsTransformer(ThreeWindingsTransformer transformer, ThreeWindingsTransformer.Side side) {
+                branchCount[0]++;
+            }
+
+            @Override
+            public void visitDanglingLine(DanglingLine danglingLine) {
+                branchCount[0]++;
+            }
+        });
+        return branchCount[0];
+    }
+
+    @Override
+    public void export(Network network, Properties parameters, DataSource dataSource, Reporter reporter) {
+        Objects.requireNonNull(network);
+        Objects.requireNonNull(dataSource);
+        Objects.requireNonNull(reporter);
+
+        MatpowerModel model = new MatpowerModel(network.getId());
+        model.setBaseMva(BASE_MVA);
+        model.setVersion(FORMAT_VERSION);
+
+        Context context = new Context();
+        boolean hasSlack = network.getBusView().getBusStream().anyMatch(MatpowerExporter::hasSlackExtension);
+        if (!hasSlack) {
+            context.refBusId = network.getBusView().getBusStream()
+                    .filter(Bus::isInMainConnectedComponent)
+                    .max(Comparator.comparingInt(MatpowerExporter::getBranchCount))
+                    .orElseThrow()
+                    .getId();
+            LOGGER.debug("Matpower reference bus automatically selected: {}", context.refBusId);
+        }
+        createBuses(network, model, context);
+        createBranches(network, model, context);
+        createGenerators(network, model, context);
+
```

Comment:
Should we check for "exotic" elements not supported by the format (SVCs, HVDCs...) and throw an exception in that case ? (or warn)